src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.forEach((format, index) => {
+    dateMap[format] = dateParts[index];
+  });
+
+  if (dateMap['YY']) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
```

Comment:
This check for 'YY' in dateMap assumes that 'YY' is always followed by 'YYYY' in the `toFormat`. However, it does not account for the case where 'YY' might be in `fromFormat` but 'YYYY' is not in `toFormat`. You should consider adding a condition to check if 'YYYY' is required in the `toFormat` before converting 'YY' to 'YYYY'.